assembly/std/arguments.ts

```diff
@@ -0,0 +1,165 @@
+import {Address} from './address';
+import {ByteArray} from '@massalabs/as/assembly/byteArray';
+
+/**
+ * Args for remote function call
+ */
+export class Args {
+  private offset: i64 = 0;
+  private argsString: string = '';
+
+  /**
+   *
+   * @param {string} argsString
+   * @param {Array<string>} types
+   */
+  constructor(argsString: string = '') {
+    this.argsString = argsString;
+  }
+
+  /**
+   *
+   * @return {string} the serialized string
+   */
+  serialize(): string {
+    return this.argsString;
+  }
+
+  // getters
+
+  /**
+   *
+   * @return {Address} the address
+   */
+  nextAddress(): Address {
+    const address = new Address();
+    this.offset = address.fromStringSegment(
+      this.argsString,
+      this.offset as i32,
+    );
+    return address;
+  }
+
+  /**
+   *
+   * @return {string} the string
+   */
+  nextString(): string {
+    let offset: i32 = this.offset as i32;
+    const length = u8(this.argsString.codePointAt(offset));
+    const result = this.argsString.slice(offset + 1, offset + length + 1);
+    this.offset = offset + length + 1;
+    return result;
+  }
+
+  /**
+   * @return {u64}
+   */
+  nextU64(): u64 {
+    const byteArray = this.fromByteString(this.argsString);
+    const amount = this.toU64(byteArray, this.offset as u8);
+    this.offset += 8;
+    return amount;
+  }
+
+  /**
+   * @return {i64}
+   */
+  nextI64(): i64 {
+    const byteArray = this.fromByteString(this.argsString);
+    const amount = this.toI64(byteArray, this.offset as u8);
+    this.offset += 8;
+    return amount;
+  }
+
+  // Setter
+
+  /**
+   *
+   * @param {T} arg the argument to add
+   *
+   * @return {Args} the modified Arg instance
+   */
+  add<T>(arg: T): Args {
+    if (arg instanceof Address) {
+      this.argsString = this.argsString.concat(arg.toStringSegment());
+    } else if (arg instanceof String || typeof arg == 'string') {
+      const str: string = arg.toString();
+      this.argsString = this.argsString.concat(
+        String.fromCharCode(u8(str.length)).concat(str as string),
+      );
+    } else if (arg instanceof u64) {
+      this.argsString = this.argsString.concat(
+        ByteArray.fromU64(arg as u64).toByteString(),
+      );
+    } else if (arg instanceof i64 || typeof arg == 'number') {
+      this.argsString = this.argsString.concat(
+        ByteArray.fromI64(arg as i64).toByteString(),
+      );
+    } else {
+      throw new Error('unsupported type: ' + typeof arg);
```

Comment:
If you throw, you stop the execution of the SC. There is no catch possibility right now in AS.

Reply:
okay so I've removed the throw, so no warnings, just do nothing in that case